commercetools/commercetools-sdk-java-api/src/main/java/com/commercetools/api/models/product/ProductProjectionMixin.java

```diff
@@ -0,0 +1,45 @@
+
+package com.commercetools.api.models.product;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public interface ProductProjectionMixin extends ProductDataLike {
```

Comment:
I added the extension because I had the error `interface com.commercetools.api.models.product.ProductProjection inherits abstract and default for getVariant(long) from types com.commercetools.api.models.product.ProductDataLike and com.commercetools.api.models.product.ProductProjectionMixin`
If you see I'm having the same error.

Reply:
I think it's better to add the methods here directly to the ProductDataLike instead

Reply:
I'm adding the changes I hope I get them right ;)